code/go/0chain.net/smartcontract/minersc/handler.go

```diff
@@ -79,19 +81,101 @@ func (msc *MinerSmartContract) GetNodepoolHandler(ctx context.Context, params ur
 }
 
 func (msc *MinerSmartContract) GetMinerListHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
-	allMinersList, err := msc.GetMinersList(balances)
+
+	var (
+		activeString  = params.Get("active")
+		hasQuery      bool
+		allMinersList = new(MinerNodes)
+		minerQuery    event.Miner
+		miners        []event.Miner
+		err           error
+	)
+
+	if balances.GetEventDB() != nil {
+		if activeString != "" && activeString != "0" {
+			active, err := strconv.Atoi(activeString)
+			if err != nil {
+				return "", fmt.Errorf("cannot parse active param (%v): %v", activeString, err)
+			}
+
+			switch active {
+			case activeNodes:
+				minerQuery = event.Miner{Active: true}
+			case inactiveNodes:
+				minerQuery = event.Miner{Active: false}
+			default:
+				return "", fmt.Errorf("invalid active param: %v. Possible values (-1, 1)", active)
+			}
+			hasQuery = true
+
+		}
+
+		if hasQuery {
+			miners, err = balances.GetEventDB().GetMinersFromQuery(&minerQuery)
+		} else {
+			miners, err = balances.GetEventDB().GetMiners()
+		}
+	} else {
+		err = errors.New("event db is not initialized")
+	}
+
 	if err != nil {
 		return "", common.NewErrInternal("can't get miners list", err.Error())
 	}
+
+	for _, miner := range miners {
+		allMinersList.Nodes = append(allMinersList.Nodes, minerTableToMinerNode(&miner))
+	}
 	return allMinersList, nil
 }
 
 const cantGetShardersListMsg = "can't get sharders list"
 
 func (msc *MinerSmartContract) GetSharderListHandler(ctx context.Context, params url.Values, balances cstate.StateContextI) (interface{}, error) {
-	allShardersList, err := getAllShardersList(balances)
+
+	var (
+		activeString    = params.Get("active")
+		hasQuery        bool
+		allShardersList = new(MinerNodes)
+		sharderQuery    event.Sharder
+		sharders        []event.Sharder
+		err             error
+	)
+
+	if balances.GetEventDB() != nil {
```

Comment:
As above, we can't use GetEvents in smart contracts. This method should not be available and this block can't exist..